alg/teca_regional_moisture_flux.cxx

```diff
@@ -0,0 +1,690 @@
+
+            auto [spivt_u, pivt_u] = get_cpu_accessible<CTT_DATA>(ivt_u);
+            auto [spivt_v, pivt_v] = get_cpu_accessible<CTT_DATA>(ivt_v);
+
+            auto [div, pdiv] = New<TT_DATA>(nelem, NT_DATA());
+            auto [cint, pcint] = New<TT_DATA>(nelem, NT_DATA());
+
+            // divergance
+            divergence(pdiv, pivt_u, pivt_v, ptheta, pphi, psin_phi, nlon, nlat);
+
+            // integrate cell by cell
+            integrate(pcint, pdiv, ptheta, pphi, nlon, nlat);
+
+            NESTED_VARIANT_ARRAY_DISPATCH(region_mask.get(), _MASK,
+
+                auto [spreg_mask, preg_mask] = get_cpu_accessible<CTT_MASK>(region_mask);
+
+                // reduce/sum
+                sum_if(moisture_flux, pcint, preg_mask, nelem);
+
+                NT_DATA Re = 6378100.;
+                moisture_flux *= -Re;
+
+                return 0;
+                )
+                TECA_ERROR("Unsupported mask data type " << region_mask->get_class_name())
+                return -1;
+            )
+            TECA_ERROR("Unsupported IVT data type " << ivt_u->get_class_name())
+            return -1;
+        )
+        TECA_ERROR("Unsupported coordinate data type " << lon->get_class_name())
+        return -1;
+}
+}
+
+
+
+
+#if defined(TECA_HAS_CUDA)
+// CUDA codes
+namespace cuda_gpu
+{
+
+
+// **************************************************************************
+template <typename coord_t>
+__global__
+void transform_coordinates(coord_t *theta, coord_t *phi, coord_t *sin_phi,
+    const coord_t *lon, const coord_t *lat, unsigned long nlon, unsigned long nlat)
+{
+    coord_t rad_deg = M_PI / 180.0;
+
+    unsigned long q = teca_cuda_util::thread_id_to_array_index();
+
+    if (q < nlon)
+    {
+        theta[q] = lon[q] * rad_deg;
+    }
+
+    if (q < nlat)
+    {
+        coord_t phi_q = ( 90.0 - lat[q] ) * rad_deg;
+
+        phi[q] = phi_q;
+        sin_phi[q] = sin( phi_q );
+    }
+}
+
+// **************************************************************************
+template <typename coord_t, typename data_t>
+__global__
+void divergence(data_t *div, const data_t *ivt_u, const data_t *ivt_v,
+    const coord_t *theta, const coord_t *phi, const coord_t *sin_phi,
+    unsigned long nlon, unsigned long nlat)
+{
+    unsigned long q = teca_cuda_util::thread_id_to_array_index();
+
+    unsigned long i = q % nlon;
+    unsigned long j = q / nlon;
+
+    if ((i < 1) || (i >= nlon - 1) || (j < 1) || (j >= nlat - 1))
+        return;
+
+    // div = d / d theta ( ivt_theta )
+    //     + d / d phi ( ivt_phi sin(phi) )
+
+    // forward difference. using a centered difference here resulted in
+    // a larger overall error.
+    data_t dthe = theta[i + 1] - theta[i];
+    data_t dphi = phi[j + 1] - phi[j];
+
+    data_t divt_the = ivt_u[q + 1] - ivt_u[q];
+    data_t divtsin_phi =  ivt_v[q + nlon] * sin_phi[j + 1] - ivt_v[q] * sin_phi[j];
+
+    div[q] = divt_the / dthe + divtsin_phi / dphi;
```

Comment:
should there be a (1/(r_earth*sin_phi[j])) multiplication here?  If it's somehow taken care of elsewhere in the code, it might be worth adding comments here about why/how

Reply:
> should there be a (1/(r_earth*sin_phi[j])) multiplication here? If it's somehow taken care of elsewhere in the code, it might be worth adding comments here about why/how

explained above.